File: runners/java/src/test/java/dev/openfga/sdk/conformance/hooks/Hooks.java
package dev.openfga.sdk.conformance.hooks;

import dev.openfga.sdk.conformance.support.TestContext;
import io.cucumber.java.Before;
import io.cucumber.java.After;
import io.cucumber.java.Scenario;

public class Hooks {
    private final TestContext testContext;

    public Hooks(TestContext testContext) {
        this.testContext = testContext;
    }

    @Before
    public void beforeScenario() {
        testContext.reset();
    }

    @After
    public void afterScenario(Scenario scenario) {
        if (scenario.isFailed()) {
            System.out.println("Scenario failed: " + scenario.getName());
            if (testContext.getLastError() != null) {
                System.out.println("Last error: " + testContext.getLastError().getMessage());
            }
        }
    }
}

File: runners/dotnet/Hooks/Hooks.cs
using TechTalk.SpecFlow;
using OpenFga.Sdk.Conformance.Support;

namespace OpenFga.Sdk.Conformance.Hooks
{
    [Binding]
    public sealed class Hooks
    {
        private readonly TestContext _testContext;

        public Hooks(TestContext testContext)
        {
            _testContext = testContext;
        }

        [BeforeScenario]
        public void BeforeScenario()
        {
            _testContext.Reset();
        }

        [AfterScenario]
        public void AfterScenario(ScenarioContext scenarioContext)
        {
            if (scenarioContext.TestError != null)
            {
                Console.WriteLine($"Scenario failed: {scenarioContext.ScenarioInfo.Title}");
                if (_testContext.LastError != null)
                {
                    Console.WriteLine($"Last error: {_testContext.LastError.Message}");
                }
            }
        }
    }
}
